### ISSUE TITLE
Edit page header redesign

### ISSUE THREAD

**Issue by desigonz** [cityofaustin/techstack#2827]
### As a content editor I want the header on an edit page to be clear and give me all of the controls I need so that I can edit, save, view a live page, and preview a draft.

**Is your feature request related to a problem? Please describe.**
This design addresses things like #2580, #2286, #2735.

**Describe the solution you'd like**
A revamped header that splits page editing away from notes and SEO; nests translations within a dropdown; indicates what page type you're editing; and surfaces things like the status of the page and allows you to view it live.
![Screen Shot 2019-08-23 at 8 32 09 AM](https://user-images.githubusercontent.com/3790067/63596339-a2d71080-c580-11e9-9c31-7260ca0a60bd.png)

**Acceptance criteria**
- [ ] Header matches mock,
  - [x] EXCEPT the SEO tab. That's a placeholder for post-MVP 
- [ ] Page type within the "edit" string—so now instead of just saying "Edit," a service page would say "Edit service page," guide would say "Edit guide," etc.
- [x] Secondary header bar with:
  - [x] Language is selected via a dropdown menu per mock
  - [x] Author notes within their own tab (labeled "Notes")
  - [x] Revisions
  - [ ] Status of page ("live," "live + draft", or "draft)
    - [ ] "Live" is a hyperlink to the live page.
    - [ ] Post-MVP: "Draft" links to the most recent draft??

**Comment by toribr** [comment]
Agree there should be parity but there is not. Except for "copy" which is available on pages but not in edit mode. I don't have much info about copy and how/whether it is used, so maybe leave that for now?

**Comment by easherma** [comment]
Comparing to the default header:
![defaultheader](https://user-images.githubusercontent.com/10539855/65282764-5806d500-dafb-11e9-8f01-16bf5591ef06.png)

We are exposing most of this functionality, so I'm going to explore whether we can re-use and reskin the existing header and avoid having to maintain a custom one but we shall see.

**Comment by easherma** [comment]
@toribr is the intention for the language dropdown that it the value seen updates with the current language selected, and the dropdown has the list of all languages?

**Comment by toribr** [comment]
@easherma Yes.

**Comment by easherma** [comment]
I have all the required elements in place and functioning. 

Next is the styling and layout. I increased the estimate because we're going to need to refactor existing css for this header and come up with something more maintainable, may need to revisit with design. 

Some notes for questions
Indication of active tab?
How will the sidebar preview come into play here?
Is the secondary menu bar fixed on the page like the primary one?
We're merging tabs along with status in a way that is a little awkward. 
We're combing page status/information with actions in a way that doesn't seem super logical.

**Comment by desigonz** [comment]
![image](https://user-images.githubusercontent.com/5839508/65453754-745f8600-de09-11e9-8c0c-92451900334f.png)

**Comment by easherma** [comment]
https://joplin-pr-2827-edit-page-heade.herokuapp.com/admin/login/ in progress

**Comment by desigonz** [comment]
Made a video using the new edit page header:
https://drive.google.com/open?id=1fqA9UHM6CipOQ3uIbnbBe7tLDiiCFY4e

Notes from @easherma on Slack:

> there a couple known issues:
> * FOUC was present before but is even more obvious now, so that ideally will get addressed
> * background color for the dropdown isn't consistent when you click away from content tab
> * hover highlighting over the various elements is a lil weird

I turned those into acceptance criteria to measure against:
- [ ] Minimize flash of unstyled content (I don't think I actually saw it though)
- [ ] Background color for language name and dropdown arrow should be the same
  - [ ] Both should be inactive when on "notes," "SEO," or "Settings" 

Not really sure what's weird about the highlighting.

@toribr, I'm also thinking this should be acceptance criteria, but correct me if I'm wrong:
- [ ] Interaction with dropdown from a non-content tab is as follows:
  1. Go to a non-content tab (ex. Notes)
  2. Click on drop-down arrow
  3. Select a language (ex. Spanish).
  4. **What I expect to happen:** Spanish content appears on page and is highlighted in the tab title. (What's currently happening is that Spanish will appear in the tab title but Notes persists on page.)

Another question for @toribr: For Joplin, we know we don't need to make it fully responsible for mobile. But at what browser breakpoints do we need to design for? Currently, if your browser windows is around 900 pixels or less, the edit page header breaks down. But a bunch of other things across Joplin don't display well at this size, so my recommendation is to backlog this and revisit later.

**Comment by toribr** [comment]
You are correct. This prototype i made shows the correct interaction (no styling, tho.)
https://pirijz.axshare.com/#g=1&p=home&c=1
I don't have permission to view the video you made.

I don't think it makes much sense to worry about browser breakpoints with so much refactoring in our future. We should make sure it looks ok on laptops and bigger.

**Comment by desigonz** [comment]
Sounds good! @easherma, let me know if there is anything else you have questions about.

**Comment by desigonz** [comment]
Assigning @toribr to review. Compile feedback here, and then I'll work on dividing feedback into different issues based on what's needed to merge this new feature versus what to do in the coming sprints. 

Tori, here are a few things I'm noticing:

The style change for the "Live" and "Draft" buttons is also affecting the buttons on the main content listing page. Left is this new PR branch; right is staging.
![image](https://user-images.githubusercontent.com/5839508/65975116-c456d200-e433-11e9-951f-01b4cae30b4e.png)

When there's a longer page title that breaks to two lines, it overlaps the tabs and links on the second bar. I would expect the bar to be pushed down to accommodate for the two-line title.

Uneven padding on this tab. @toribr or @chasechenevert, can y'all specify what the padding should be?
![image](https://user-images.githubusercontent.com/5839508/65975465-46df9180-e434-11e9-9ac3-d8deb47cf1bc.png)

![image](https://user-images.githubusercontent.com/5839508/65975601-77273000-e434-11e9-9386-ec962b8e45f6.png)

Share button isn't producing the copy to clipboard thing:
![2827-share](https://user-images.githubusercontent.com/5839508/65975918-fd437680-e434-11e9-9d56-94a36bbf31da.gif)

There's a little flash of the word "content" in the tab before it changes to "English"
![2827-content](https://user-images.githubusercontent.com/5839508/65976550-e5202700-e435-11e9-95b6-942cc5c488eb.gif)

@easherma I want to test this acceptance criteria: `"Live" is a hyperlink to the live page.` but I don't think there's a Janis build for this Joplin PR branch. How can I test it?

**Comment by toribr** [comment]
-[  ] remove radius on first tab {border-radius: 3px 3px 0px 0px;}
- [  ] make hover not exceed bar 
![Screen Shot 2019-10-01 at 12 53 04 PM](https://user-images.githubusercontent.com/3790067/65987417-5f0edb00-e44b-11e9-98ed-366b8b99abfe.png)

**Comment by desigonz** [comment]
Acceptance criteria for pushing this to production:
- [ ] No flash of "content" in the tab before it changes to "English" (see gif above)
- [ ] When page title breaks to two lines, the tab row still displays as expected
- [ ] Share button generates clipboard link as expected

Waiting on @toribr to share thoughts on the new "live" button style, which is also affecting the main page.

Also made this issue for all of the style fixes: https://github.com/cityofaustin/techstack/issues/3087 which we should do this next sprint

**Comment by briaguya** [comment]
@desigonz I think the flash of "content" is related to this: https://github.com/cityofaustin/techstack/issues/3081

@easherma confirm?

**Comment by easherma** [comment]
@briaguya correct! It also happens when we actually filter out the languages, even on prod right now if you watch closely or have a slow loading page, you can see all the language fields before they are hidden.
